### PR TITLE
add io.github.erkin.ponysay

### PR DESCRIPTION
**Pull request by SharkyRawr**
Add ponysay commandline tool from https://github.com/erkin/ponysay

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
The application should use a proper app-id.

**Comment by SharkyRawr** [comment]
> The application should use a proper app-id.

trying to figure out the naming scheme, would `io.github.erkin.ponysay` be more appropriate? Thanks!

**Comment by bilelmoussaoui** [comment]
> > The application should use a proper app-id.
> 
> trying to figure out the naming scheme, would `io.github.erkin.ponysay` be more appropriate? Thanks!

That's better yes, I would check if upstream has one first. Also the wiki should help https://github.com/flathub/flathub/wiki/App-Requirements#application-id

**Comment by SharkyRawr** [comment]
Checked flathub but could not find anything close to the namespace or appid so I changed it accordingly. Thanks for your help! :heart:

File: io.github.erkin.ponysay.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+sdk: org.freedesktop.Sdk
+command: /app/bin/ponysay
+modules:
+  - name: ponysay
+    buildsystem: simple
+    build-commands:
+      - ./setup.py --prefix /app --freedom=partial install
+    sources:
+      - type: git
+        url: https://github.com/erkin/ponysay.git
```
Comment (nedrichards) [inline io.github.erkin.ponysay.yaml:13]: this needs to be a specific commit or release, you shouldn't just point at master as the build can't be exactly reproduced then.

**Comment by nedrichards** [comment]
bot, build io.github.erkin.ponysay

**Comment by nedrichards** [comment]
Are you part of the upstream project? If not is there a ticket you can point to asking if upstream would like to be involved? (do file one if not), similarly you should offer additional changes like the appdata to upstream as well so they can be used by all Linux distribution systems.

**Comment by nedrichards** [comment]
Also: looks like the appdata isn't getting installed correctly inside the filesystem of the container.

**Comment by LeandroStanger** [comment]
bot, build io.github.erkin.ponysay

File: io.github.erkin.ponysay.yaml (hunk 1 above)
Comment (SharkyRawr) [reply io.github.erkin.ponysay.yaml:13]: Added latest tag and commit hash to yaml.

**Comment by SharkyRawr** [comment]
> Also: looks like the appdata isn't getting installed correctly inside the filesystem of the container.

Whoops, first time trying to make a flatpak. :) Forgot about that. _Should_ be fixed now.

**Comment by SharkyRawr** [comment]
Created an issue with the upstream project: https://github.com/erkin/ponysay/issues/302

**Comment by nedrichards** [comment]
bot, build io.github.erkin.ponysay

**Comment by SharkyRawr** [comment]
Reduced `<developer_name>` content length below 60 characters.

**Comment by nedrichards** [comment]
bot, build io.github.erkin.ponysay

**Comment by SharkyRawr** [comment]
Added a 128x128 icon and post-install step:
```yaml
- install -Dm644 ponysay_128x128.png /app/share/app-info/icons/flatpak/128x128/io.github.erkin.ponysay.png
```
This should hopefully satisfy the build error. (?)
```
zgrep "<icon type=\"remote\">" builddir/*/share/app-info/xmls/io.github.erkin.ponysay.xml.gz || test -f builddir/*/share/app-info/icons/flatpak/128x128/io.github.erkin.ponysay.png
```

**Comment by bilelmoussaoui** [comment]
> Added a 128x128 icon and post-install step:
> 
> ```yaml
> - install -Dm644 ponysay_128x128.png /app/share/app-info/icons/flatpak/128x128/io.github.erkin.ponysay.png
> ```
> 
> This should hopefully satisfy the build error. (?)
> 
> ```
> zgrep "<icon type=\"remote\">" builddir/*/share/app-info/xmls/io.github.erkin.ponysay.xml.gz || test -f builddir/*/share/app-info/icons/flatpak/128x128/io.github.erkin.ponysay.png
> ```

It's a CLI application, you don't have to ship an icon. You can disable that by adding a `flathub.json` file that contains` skip-icons-check: true` see https://discourse.flathub.org/t/a-128x128-icon-is-required-now/386/4

**Comment by Eonfge** [comment]
As the maker and publisher of [Cowsay](https://flathub.org/apps/details/org.gnome.gitlab.Cowsay) here on Flathub, I can only encourage such variety in speaking animal mascots.
```
 _________________
< outstanding job >
 -----------------
   \        \
    \        \
     \       _\^
      \    _- oo\
           \---- \______
                 \       )\
                ||-----||  \
                ||     ||

```

**Comment by nedrichards** [comment]
bot, build io.github.erkin.ponysay

**Comment by CodeMouse92** [comment]
Is there a reason that the app is not found on Flathub, then? https://flathub.org/apps/details/io.github.erkin.ponysay says "App not found"

**Comment by SharkyRawr** [comment]
@CodeMouse92 maybe because it's marked as a console application? Curious to know myself.

**Comment by bilelmoussaoui** [comment]
Yes. Console applications are not supported on the frontend yet.

**Pull request closed without merging**